### PR TITLE
T 2 create left menu markup

### PR DESCRIPTION
**Pull request by ShevchenkoPavel**
https://github.com/ShevchenkoPavel/chat/projects/1#card-27081967

### CODE REVIEW

**Comment by ShevchenkoPavel** [comment]
https://github.com/ShevchenkoPavel/chat/projects/1#card-27081967

File: src/layout/layout.css [hunk 1]
```diff
@@ -0,0 +1,10 @@
+.layout {
```
Comment (oTkPoBeHuE) [inline src/layout/layout.css]: Use  [react-bootstrap grid](https://react-bootstrap.netlify.com/layout/grid/). it now supports flexblox.

File: src/layout/layout.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import './layout.css';
+import { UsersBar } from '../users_bar/users_bar';
+// import { get } from '../queries/fetcher';
```
Comment (oTkPoBeHuE) [inline src/layout/layout.tsx]: Please, delete comment. We have history from git.

File: src/layout/layout.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  users: Array<UserType>;
+};
+
+export class Layout extends React.Component<Props, State> {
+  state = {
+    users: []
+  };
+
+  componentDidMount(): void {
+    // get('localhost/public/mockapi/users.js');
```
Comment (oTkPoBeHuE) [inline src/layout/layout.tsx]: Please, delete comment. We have history from git.

File: src/queries/fetcher.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  }
+  return Promise.reject(new Error(response.statusText));
+}
+
+function getResponseJson(response: Response) {
+  return response.json();
+}
+
+export function get(url: string) {
+  return fetch(url, {
```
Comment (oTkPoBeHuE) [inline src/queries/fetcher.ts]: Please, rewrite to async/await try/catch.

File: src/user/user.css [hunk 5]
```diff
@@ -0,0 +1,13 @@
+.user {
```
Comment (oTkPoBeHuE) [inline src/user/user.css]: Use grid system from react-bootstrap

File: src/users_bar/users_bar.css [hunk 6]
```diff
@@ -0,0 +1,17 @@
+.users-bar {
+    min-width: 20%;
+    box-shadow: 1px 0 5px grey;
+}
+
+.users-bar__header {
+    background-color: rgb(0, 188, 212);
+    font-size: x-large;
+    /*text-align: center;*/
```
Comment (oTkPoBeHuE) [inline src/users_bar/users_bar.css]: Delete comment

File: src/users_bar/users_bar.css [hunk 7]
```diff
@@ -0,0 +1,17 @@
+.users-bar {
+    min-width: 20%;
+    box-shadow: 1px 0 5px grey;
+}
+
+.users-bar__header {
+    background-color: rgb(0, 188, 212);
```
Comment (oTkPoBeHuE) [inline src/users_bar/users_bar.css]: Rename all `.css` to `.sass`/`scss`. and move  cvolor to const

File: src/layout/layout.tsx [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+import { UsersBar } from '../users_bar/users_bar';
+import MockUsers from '../mock/users.json';
+import { UserType } from '../user/user';
+
+type Props = {};
+type State = {
+  users: Array<UserType>;
+};
+
+export class Layout extends React.Component<Props, State> {
```
Comment (oTkPoBeHuE) [inline src/layout/layout.tsx]: Rewrite to functional component with react [React hooks](https://ru.reactjs.org/docs/hooks-reference.html)

File: src/layout/layout.css [hunk 9]
```diff
@@ -0,0 +1,10 @@
...
+    display: flex;
+    flex-direction: row;
+    justify-content: space-between;
+    /*background-color: antiquewhite;*/
```
Comment (oTkPoBeHuE) [inline src/layout/layout.css]: delete comment

File: src/layout/layout.css (hunk 1 above)
Comment (ShevchenkoPavel) [reply src/layout/layout.css]: fixed

File: src/layout/layout.tsx (hunk 2 above)
Comment (ShevchenkoPavel) [reply src/layout/layout.tsx]: fixed

File: src/layout/layout.tsx (hunk 3 above)
Comment (ShevchenkoPavel) [reply src/layout/layout.tsx]: fixed

File: src/queries/fetcher.ts (hunk 4 above)
Comment (ShevchenkoPavel) [reply src/queries/fetcher.ts]: not ready yet

File: src/user/user.css (hunk 5 above)
Comment (ShevchenkoPavel) [reply src/user/user.css]: fixed

File: src/users_bar/users_bar.css (hunk 6 above)
Comment (ShevchenkoPavel) [reply src/users_bar/users_bar.css]: done

File: src/users_bar/users_bar.css (hunk 7 above)
Comment (ShevchenkoPavel) [reply src/users_bar/users_bar.css]: done

File: src/layout/layout.tsx (hunk 8 above)
Comment (ShevchenkoPavel) [reply src/layout/layout.tsx]: component deleted

File: src/layout/layout.css (hunk 9 above)
Comment (ShevchenkoPavel) [reply src/layout/layout.css]: fixed

File: src/components/user/user.tsx [hunk 10]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import './user.scss';
+
+export type UserType = {
```
Comment (oTkPoBeHuE) [inline src/components/user/user.tsx]: Rename to `Props` or `UserProps`.  Don't  export this props. If you need Type User in other files, move it to special file and export in user.tsx:

``` tsx
import { UserType } from 'types.ts';

type UserProps = {
   user: UserType;
}
```

File: src/components/users_bar/users_bar.tsx [hunk 11]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import './users_bar.scss';
+import { User } from '../user/user';
+
+interface UserType {
```
Comment (oTkPoBeHuE) [inline src/components/users_bar/users_bar.tsx]: Move `UserType` to separate file. See above.

File: src/components/users_bar/users_bar.tsx [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+}
+
+export type UsersBarProps = {
+  users?: UserType[];
+};
+
+export const UsersBar = ({ users }: UsersBarProps) => {
+  return (
+    <div className="users-bar">
+      <div className="users-bar__header">users</div>
```
Comment (oTkPoBeHuE) [inline src/components/users_bar/users_bar.tsx]: Move all string constants to separate file.
 _lang/US.ts_:

``` tsx
export const lang = {
   USERS: 'users'
};
```

File: src/pages/chat/chat.tsx [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+import MockUsers from '../../utils/mock/users.json';
+import { UserType } from '../../components/user/user';
+import { Container, Row, Col } from 'react-bootstrap';
+
+type Props = {};
+type State = {
+  users: Array<UserType>;
+};
+
+export class Chat extends React.Component<Props, State> {
```
Comment (oTkPoBeHuE) [inline src/pages/chat/chat.tsx]: Please rewrite  to functional component with React hooks

File: src/components/users_bar/users_bar.tsx [hunk 14]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import './users_bar.scss';
+import { User } from '../user/user';
+
+interface UserType {
+  name: string;
+}
+
+export type UsersBarProps = {
+  users?: UserType[];
```
Comment (oTkPoBeHuE) [inline src/components/users_bar/users_bar.tsx]: If there are no users you can pass empty array. User should not be `undefined`.

File: src/queries/fetcher.ts (hunk 4 above)
Comment (ShevchenkoPavel) [reply src/queries/fetcher.ts]: done

**Pull request merged**